tags/tags-impl/impl/src/java/org/sakaiproject/tags/impl/rest/TagServiceEntityProvider.java

```diff
@@ -60,7 +64,9 @@ public class TagServiceEntityProvider implements EntityProvider, AutoRegisterEnt
 
     protected DeveloperHelperService developerHelperService;
     private EntityProviderManager entityProviderManager;
-    private  SessionManager sessionManager = (SessionManager) ComponentManager.get("org.sakaiproject.tool.api.SessionManager");
+    private SessionManager sessionManager = (SessionManager) ComponentManager.get("org.sakaiproject.tool.api.SessionManager");
+
+    private AssignmentService assignmentService = (AssignmentService) ComponentManager.get("org.sakaiproject.assignment.api.AssignmentService");;
```

Comment:
It propbably makes sense to use the entity manager vs the assignment service directly as your only retrieving assignments.
Also it would be ideal to migrate away from using the ComponentManager directly

Reply:
I changed the approach deleting the use of assignment service, the endpoint could be used for other tools.